### PR TITLE
add arg table for ReferenceArea stories

### PR DESCRIPTION
**Pull request by lukask-proxora**
## Related Issue

#3432 

## Screenshots (if appropriate):
Screenshot from 2023-03-26 17-57-49
Screenshot from 2023-03-26 17-57-43
Screenshot from 2023-03-26 17-57-31
Screenshot from 2023-03-26 17-57-16
Screenshot from 2023-03-26 17-57-07
Screenshot from 2023-03-26 17-56-48

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by Yilun-Sun** [comment]
Hi @lukask-proxora! Please rebase to `master` to avoid CI failing on type errors. Thanks!

**Comment by ckifer** [comment]
+1 to the above, thanks

File: storybook/stories/API/props/AnimationProps.ts [hunk 1]
```diff
@@ -9,24 +9,37 @@ export const animationBegin = {
   description: 'Specifies when the animation should begin, the unit of this option is ms.',
   type: { name: 'number' },
   defaultValue: 0,
```
Comment (nikolasrieble) [inline storybook/stories/API/props/AnimationProps.ts:11]: Now the defaultValue seems to be defined twice. Is that necessary?

**Review by nikolasrieble — APPROVED** [review]
Looks great, please remove the redundant duplicate definition of the defaultValue in the AnimationProps. Not blocking though.

**Comment by ckifer** [comment]
merging in for now, thanks!

File: storybook/stories/API/props/AnimationProps.ts (hunk 1 above)
Comment (lukask-proxora) [reply storybook/stories/API/props/AnimationProps.ts:11]: Hi @nikolasrieble, without the `defaultValue` in the `table` prop, the default value in the args table is not set to the manually set `defaultValue` of the argType, but only to the inferred default value. 

According to https://github.com/storybookjs/storybook/issues/11983#issuecomment-896790639 (which also refers to https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#no-longer-inferring-default-values-of-args):

> To affect the Default column in the ArgsTable, you should set the argTypes.<field>.table.defaultValue field. The value should be an object with the keys summary and optionally details, which are both strings.

**Pull request merged**